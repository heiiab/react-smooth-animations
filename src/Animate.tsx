import * as React from 'react';
import Base from './Base';
import { AnimateProps } from './utils/interfaces';

import FadeIn from './Animations/FadeIn';
import FadeOut from './Animations/FadeOut';
import FadeInUp from './Animations/FadeInUp';
import FadeOutUp from './Animations/FadeOutUp';
import FadeInLeft from './Animations/FadeInLeft';
import FadeInDown from './Animations/FadeInDown';
import FadeInRight from './Animations/FadeInRight';
import FadeOutLeft from './Animations/FadeOutLeft';
import FadeOutDown from './Animations/FadeOutDown';
import FadeOutRight from './Animations/FadeOutRight';

interface AnimateInterface extends React.FC<AnimateProps> {
  FadeIn: typeof FadeIn;
  FadeOut: typeof FadeOut;
  FadeInUp: typeof FadeInUp;
  FadeOutUp: typeof FadeOutUp;
  FadeInLeft: typeof FadeInLeft;
  FadeInDown: typeof FadeInDown;
  FadeInRight: typeof FadeInRight;
  FadeOutLeft: typeof FadeOutLeft;
  FadeOutDown: typeof FadeOutDown;
  FadeOutRight: typeof FadeOutRight;
}

const Animate: AnimateInterface = ({ children, ...restProps }) => <Base {...restProps}>{children}</Base>;

Animate.FadeIn = FadeIn;
Animate.FadeOut = FadeOut;
Animate.FadeInUp = FadeInUp;
Animate.FadeOutUp = FadeOutUp;
Animate.FadeInLeft = FadeInLeft;
Animate.FadeInDown = FadeInDown;
Animate.FadeInRight = FadeInRight;
Animate.FadeOutLeft = FadeOutLeft;
Animate.FadeOutDown = FadeOutDown;
Animate.FadeOutRight = FadeOutRight;

export default Animate;
