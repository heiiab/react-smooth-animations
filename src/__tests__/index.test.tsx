import Animate from '..';

describe('react-smooth-animations', () => {
  afterAll(() => {});

  it('exports module correctly', () => {
    expect(Object.keys(Animate)).toMatchSnapshot();
  });
});
