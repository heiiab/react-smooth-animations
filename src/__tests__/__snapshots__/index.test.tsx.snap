// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-smooth-animations exports module correctly 1`] = `
Array [
  "FadeIn",
  "FadeInUp",
  "FadeInLeft",
  "FadeInDown",
  "FadeInRight",
  "FadeOut",
  "FadeOutUp",
  "FadeOutLeft",
  "FadeOutDown",
  "FadeOutRight",
  "SlideInUp",
  "SlideInLeft",
  "SlideInDown",
  "SlideInRight",
  "SlideOutUp",
  "SlideOutLeft",
  "SlideOutDown",
  "SlideOutRight",
  "ZoomIn",
  "ZoomInUp",
  "ZoomInLeft",
  "ZoomInDown",
  "ZoomInRight",
  "ZoomOut",
  "ZoomOutUp",
  "ZoomOutLeft",
  "ZoomOutDown",
  "ZoomOutRight",
  "BounceIn",
  "BounceInUp",
  "BounceInLeft",
  "BounceInDown",
  "BounceInRight",
  "BounceOut",
  "BounceOutUp",
  "BounceOutLeft",
  "BounceOutDown",
  "BounceOutRight",
  "FlipInUp",
  "FlipInLeft",
  "FlipInDown",
  "FlipInRight",
  "FlipOutUp",
  "FlipOutLeft",
  "FlipOutDown",
  "FlipOutRight",
]
`;
